app-creator/client/src/widgets/tool-bar/tool-bar.ts

```diff
@@ -77,30 +80,79 @@ export class ToolBar extends LitElement {
       margin-right: var(--tight);
     }
 
+    #import-textarea {
+      width: calc(100% - 2 * var(--tight) - 44px);
+      height: 250px;
+      margin-left: 24px;
+      padding: var(--tight);
+    }
+
+    #import-button {
+      background-color: var(--primary-color);
+      color: var(--accent-color);
+      height: 30px;
+      font-size: 0.8rem;
+      border: 0.3px solid var(--accent-color);
+    }
+
     #export-button {
       background-color: var(--accent-color);
       color: var(--primary-color);
       height: 30px;
       font-size: 0.8rem;
     }
+
+    #invalid-json-toast {
+      background-color: var(--validation-error-red-color);
+    }
   `;
 
+  /**
+   * Reference to the export dialog element.
+   */
+  @query('#export-dialog') exportDialog!: PaperDialogElement;
+
+  /**
+   * Reference to the import dialog element.
+   */
+  @query('#import-dialog') importDialog!: PaperDialogElement;
+
+  /**
+   * Reference to the textarea element.
+   */
+  @query('#import-textarea') importTextArea!: HTMLTextAreaElement;
+
   /**
    * Triggered when export button is clicked. It displays the paper dialog which
    * contains the serialized template string.
    */
-  openDialog() {
-    const dialog = this.shadowRoot?.querySelector('paper-dialog');
+  openExportDialog() {
     const jsonSnippetContainer = this.shadowRoot?.getElementById(
       'json-snippet'
     );
 
-    if (dialog == null || jsonSnippetContainer == null) {
+    if (this.exportDialog == null || jsonSnippetContainer == null) {
       return;
     }
     jsonSnippetContainer.textContent = this.getTemplateString(3);
 
-    (dialog as PaperDialogElement).open();
+    if ((this.exportDialog as PaperDialogElement).open) {
+      (this.exportDialog as PaperDialogElement).open();
+    }
```

Comment:
I think we could safely remove the type guard here, since the `@query` decorator has already cased `this.exportDialog` to `PaperDialogElement`, same for `this.importDialog`. Apologies about the conflicting comment earlier!